unstructured_inference/models/chipper.py

```diff
@@ -1,173 +1,502 @@
+from unstructured_inference.inference.elements import Rectangle
+from unstructured_inference.inference.layoutelement import LayoutElement
 from unstructured_inference.models.unstructuredmodel import UnstructuredElementExtractionModel
+from unstructured_inference.utils import LazyDict
 
-MODEL_TYPES = {
+MODEL_TYPES: Dict[Optional[str], Union[LazyDict, dict]] = {
     "chipper": {
-        "tokenizer_name": "xlm-roberta-large",
-        "pre_trained_model_name": "unstructuredio/ved-fine-tuning",
+        "pre_trained_model_repo": "unstructuredio/ved-fine-tuning",
+        "swap_head": False,
+        "prompt": "<s>",
+        "max_length": 1200,
+        "heatmap_h": 52,
+        "heatmap_w": 39,
+        "source": Source.CHIPPER.value,
+    },
+    "chipperv2": {
+        "pre_trained_model_repo": "unstructuredio/chipper-fast-fine-tuning",
+        "swap_head": True,
+        "prompt": "<s><s_hierarchical>",
+        "max_length": 1536,
+        "heatmap_h": 40,
+        "heatmap_w": 30,
+        "source": Source.CHIPPERV2.value,
     },
 }
 
-LABEL_MAP = [
-    "Title",
-    "Abstract",
-    "Headline",
-    "Subheadline",
-    "Text",
-    "Table",
-    "List",
-    "List-item",
-    "Page number",
-    "Header",
-    "Footer",
-    "Address",
-    "Author",
-    "Chart",
-    "Caption",
-    "Formula",
-    "Picture",
-    "Advertisement",
-    "Link",
-    "Misc",
-    "Field-Name",
-    "Value",
-    "Threading",
-    "Metadata",
-    "Form",
-]
-
-# NOTE(alan): These are the special tokens that are generated by the model and will be associated
-# with element types. The order of this particular list matters as the tokenizer vocabulary must not
-# change. Otherwise we won't correctly decode the tokens generated by the model.
-SPECIAL_TAGS = ["<sep/>"] + [
-    f"<{open_or_close}s_{label_type}>" for label_type in LABEL_MAP for open_or_close in ("", "/")
-]
-
 
 class UnstructuredChipperModel(UnstructuredElementExtractionModel):
-    required_w: int = 1248
-    required_h: int = 1664
-
     def initialize(
         self,
-        tokenizer_name: str,
-        pre_trained_model_name: str,
+        pre_trained_model_repo: str,
+        swap_head: bool,
+        prompt: str,
+        max_length: int,
+        heatmap_h: int,
+        heatmap_w: int,
+        source: str,
+        no_repeat_ngram_size: int = 10,
         auth_token: Optional[str] = os.environ.get("UNSTRUCTURED_HF_TOKEN"),
     ):
         """Load the model for inference."""
-        self.tokenizer: AutoTokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
-
-        self.processor: DonutProcessor = DonutProcessor(
-            image_processor=DonutImageProcessor(
-                do_resize=True,
-                size=(self.required_w, self.required_h),
-            ),
-            tokenizer=self.tokenizer,
+        if torch.cuda.is_available():
+            self.device = "cuda"
+        else:
+            self.device = "cpu"
```

Comment:
auto detection is nice but I would suggest we also add into init the ability to specify device. And this logic is used to either provide default or change when specified device is not available

Reply:
This would be nice not only for Chipper, since many GPUs might exist in the same system. This probably requires its own feature request.

Reply:
Added a parameter that can be passed that overrides autodetection in [6aa76ab](https://github.com/Unstructured-IO/unstructured-inference/pull/232/commits/6aa76ab0536240849efd78f7b76ddbe06f750164). We'd have to think about how a user could actually pass this in, but the framework is there.